python_app/utils.py

```
@@ -0,0 +1,44 @@
+"""Utility functions for sensor data processing."""
+
+import re
+import time
+import random
+import functools
+import itertools
+import collections
+import threading
+from typing import Any, Dict, List, Optional, Callable, Union
```

Ruff (0.15.6)

[warning] Import from `collections.abc` instead: `Callable`

Import from `collections.abc`

(UP035)